hazelcast/src/main/java/com/hazelcast/sql/SqlResult.java

```diff
@@ -53,17 +69,26 @@
      * The iterator may be requested only once.
      *
      * @return iterator
-     * @throws IllegalStateException if the method is invoked more than once
+     * @throws IllegalStateException if the method is invoked more than once or
+     *      if the {@linkplain #getResultType() result type} isn't {@link
+     *      SqlResultType#ROWS ROWS}.
      * @throws SqlException in case of an SQL-related error condition
      */
     @Nonnull
     @Override
     Iterator<SqlRow> iterator();
 
     /**
-     * Release the resources associated with the query result.
+     * Returns the result type.
+     */
+    @Nonnull
+    SqlResultType getResultType();
```

Comment:
Technically it will suffice. The problem is the UX. And my opinion is the void result communicates the intent better: there really is no result for a DDL statement. There also won't be an update count for an INSERT query executed by Jet.

Personally, I like the design used by JDBC better. They use the term "statement", not "query". When you say "query", you typically mean SELECT query. DDL statement is not a query, but you still use `query()` method to execute it. JDBC has multiple methods:
- void execute() - for DDL or for unknown query where you can determine the result type later
- ResultSet executeQuery() - if your statement resturns rows
- int executeUpdate() - if your statement is supposed to return an updated/affected row count

The last two throw if the statement result is of a wrong type.

Reply:
We can rethink our API. E.g. `SqlQuery` -> `SqlStatement`, `SqlService.query` -> `SqlService.execute`.

My idea is more about using a single boolean flag to distinguish the absence or presence of the result set. If the result set is absent, it is up to the user whether to read the "update count" or not, because it's semantics may be different depending on the command. `VOID` is hardly distinguishable from updateCount=0.

Reply:
We can also use `updateCount=-1` for jet's INSERT statements because the updated count isn't zero, but unknown.

Reply:
addressed in a new PR: https://github.com/hazelcast/hazelcast/pull/17281